src/app/space/create/deployments/deployments-donut/deployments-donut-chart/deployments-donut-chart.component.spec.ts

```diff
@@ -21,82 +23,69 @@ describe('DeploymentsDonutChartComponent', () => {
     [idled]="isIdled">
     </deployments-donut-chart>
     `
-  })
-  class TestHostComponent {
-    mini = false;
-    pods = Observable.of({ pods: [['Running', 1], ['Terminating', 1]], total: 2 });
-    desiredReplicas = 1;
-    isIdled = false;
-    colors = {
-      'Empty': '#030303', // pf-black
-      'Running': '#00b9e4', // pf-light-blue-400
-      'Not Ready': '#beedf9', // pf-light-blue-100
-      'Warning': '#f39d3c', // pf-orange-300
-      'Error': '#cc0000', // pf-red-100
-      'Pulling': '#d1d1d1', // pf-black-300
-      'Pending': '#ededed', // pf-black-200
-      'Succeeded': '#3f9c35', // pf-green-400
-      'Terminating': '#00659c', // pf-blue-500
-      'Unknown': '#f9d67a' // pf-gold-200
-    };
-
-    @ViewChild(DeploymentsDonutChartComponent)
-    childComponent: DeploymentsDonutChartComponent;
-  }
-
-  let hostComponent: TestHostComponent;
-  let fixture: ComponentFixture<TestHostComponent>;
+})
+class TestHostComponent {
+  mini = false;
+  pods = Observable.of({ pods: [['Running', 1], ['Terminating', 1]], total: 2 });
+  desiredReplicas = 1;
+  isIdled = false;
+  colors = {
+    'Empty': '#030303', // pf-black
+    'Running': '#00b9e4', // pf-light-blue-400
+    'Not Ready': '#beedf9', // pf-light-blue-100
+    'Warning': '#f39d3c', // pf-orange-300
+    'Error': '#cc0000', // pf-red-100
+    'Pulling': '#d1d1d1', // pf-black-300
+    'Pending': '#ededed', // pf-black-200
+    'Succeeded': '#3f9c35', // pf-green-400
+    'Terminating': '#00659c', // pf-blue-500
+    'Unknown': '#f9d67a' // pf-gold-200
+  };
 
-  let de: DebugElement;
-  let el: HTMLElement;
+  @ViewChild(DeploymentsDonutChartComponent)
+  childComponent: DeploymentsDonutChartComponent;
+}
 
-  beforeEach(() => {
-    TestBed.configureTestingModule({
-      declarations: [DeploymentsDonutChartComponent, TestHostComponent]
-    });
-
-    fixture = TestBed.createComponent(TestHostComponent);
-    hostComponent = fixture.componentInstance;
-
-    fixture.detectChanges();
-  });
+describe('DeploymentsDonutChartComponent', () => {
+  type Context = TestContext<DeploymentsDonutChartComponent, TestHostComponent>;
+  initContext(DeploymentsDonutChartComponent, TestHostComponent);
 
-  it('should set unique chartId', () => {
-    expect(hostComponent.childComponent.chartId).toMatch('deployments-donut-chart.*');
+  it('should set unique chartId', function (this: Context) {
+    expect(this.hostComponent.childComponent.chartId).toMatch('deployments-donut-chart.*');
```

Comment:
Now that it uses the test context, can `childComponent` be removed in favor of `this.testedDirective`?